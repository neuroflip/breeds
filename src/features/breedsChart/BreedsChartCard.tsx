import ResponsivePieChart from "./components/ResponsivePieChart/ResponsivePieChart"
import Card from "../../components/Card/Card"
import useBreedsChartCard from "./hooks/useBreedsChartCard"

const BreedsChartCard = () => {
  const { renderTitle, breedImagesPercent } = useBreedsChartCard()

  return <Card renderTitle={renderTitle}>
      <ResponsivePieChart data={ breedImagesPercent } />
    </Card>
}

export default BreedsChartCard