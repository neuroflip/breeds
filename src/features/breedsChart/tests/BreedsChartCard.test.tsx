import { describe, expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import BreedsChartCard from '../BreedsChartCard';

vi.mock('../components/ResponsivePieChart/ResponsivePieChart')
vi.mock('../../../components/Card/Card')
vi.mock('../hooks/useBreedsChartCard', () => ({
  __esModule: true,
  default: () => Promise.resolve({
    renderTitle: vi.fn(),
    breedImagesPercent: [{
      name: "breed1",
      value: 100
    }]    
  })
}));

describe('BreedsChartCard', () => {
  test('initial render', () => {
    const { container } = render(<BreedsChartCard />);

    expect(screen.getByTestId('Card')).toBeInTheDocument()
    expect(screen.getByText('ResponsivePieChart')).toBeInTheDocument()

    //snapshot
    expect(container).toMatchSnapshot()
  });
});