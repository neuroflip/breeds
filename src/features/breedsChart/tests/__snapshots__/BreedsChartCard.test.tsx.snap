// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BreedsChartCard > initial render 1`] = `
<div>
  <div
    data-testid="Card"
  >
    ResponsivePieChart
  </div>
</div>
`;
